pkg/bytecode/compiler.go

```diff
@@ -183,6 +183,9 @@ func (c *Compiler) compileBinaryExpression(expr *parser.BinaryExpression) error
 	if expr.Left.Type().Name == parser.ARRAY && expr.Right.Type().Name == parser.ARRAY && expr.Op == parser.OP_PLUS {
 		return c.emit(OpArrayConcatenate)
 	}
+	if expr.Left.Type().Name == parser.ARRAY && expr.Right.Type() == parser.NUM_TYPE {
```

Comment:
```suggestion
	if expr.Left.Type().Name == parser.ARRAY && expr.Right.Type() == parser.NUM_TYPE && expr.Op == parser.OP_ASTERISK {
```
being specific about the op doesn't hurt

Reply:
Thanks. This is arguably a bug. Fixed.